### PR TITLE
Upgrade to Rust 2021

### PR DESCRIPTION
**Pull request by TimTheBig**
### Updates
Rust edition **2018** –> **2021**
Version **2.5.2** –> **2.6.0**
msrv according to cargo-msrv **1.40.0** –> **1.60.0**, tested on x64 and arm64.

### Small changes
Added `"xdg"` as keyword in `cargo.toml`
Add Swatinem/rust-cache to `ci.yml` and `docs.yml` for CI speed up.
Add **CI**, **Doc-test**, **msrv** and **version** badges to `README.md`.
`actions/checkout` **v3** –> **v4**
Make lint workflow continue on fmt error, so clippy can also run.

### CODE REVIEW

**Comment by whitequark** [comment]
Thanks! It looks like clippy is still unhappy: https://github.com/TimTheBig/rust-xdg/actions/runs/9257836687/job/25466781313#step:6:22

**Comment by TimTheBig** [comment]
The main clippy complaint does make the code a lot cleaner so I would consider it:                                          
```rust 
self.state_home.as_ref().map(|home| home.as_path()),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using`self.state_home.as_deref()`

```
I can also make clippy ignore it if you prefer.

**Comment by whitequark** [comment]
Yeah, let's apply that.

**Comment by TimTheBig** [comment]
> Yeah, let's apply that.
Apply the lint?

File: README.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
-rust-xdg
-========
+# rust-xdg
+
+[![CI](https://github.com/whitequark/rust-xdg/actions/workflows/ci.yml/badge.svg)](https://github.com/whitequark/rust-xdg/actions/workflows/ci.yml)
+[![Documentation](https://github.com/whitequark/rust-xdg/actions/workflows/docs.yml/badge.svg)](https://github.com/whitequark/rust-xdg/actions/workflows/docs.yml)
+![Crates.io Downloads (recent)](https://img.shields.io/crates/dr/xdg?logo=rust&color=%23a55e08&link=https%3A%2F%2Fcrates.io%2Fcrates%2Fxdg)
```
Comment (whitequark) [inline README.md:5]: I'd rather not have the download count in README.

**Comment by TimTheBig** [comment]
Feel free to swash on merge.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 /target/
 /Cargo.lock
+.DS_Store
```
Comment (whitequark) [inline .gitignore:3]: This should go into your system-local `.gitignore`.

**Review by whitequark — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (TimTheBig) [reply .gitignore:3]: I was unaware of system-local `.gitignore`'s my mistake.
```suggestion
```

**Pull request merged**